test/test_downloadable_library.py

```diff
@@ -0,0 +1,175 @@
+            self.library_uuid,
+            self.library_file,
+        )
+        # 内容はdownloadable_library.jsonを元に生成されるので、内容は確認しない
+        self.assertEqual(
+            list(self.library_manger.installed_libraries().keys())[0], self.library_uuid
+        )
+
+        self.library_manger.uninstall_library(self.library_uuid)
+        self.assertEqual(self.library_manger.installed_libraries(), {})
+
+    def test_install_library(self):
+        with self.assertRaises(HTTPException) as e:
+            self.library_manger.install_library(
+                "52398bd5-3cc3-406c-a159-dfec5ace4bab", self.library_file
+            )
+        self.assertEqual(e.exception.detail, "指定された音声ライブラリが見つかりません。")
+
+        with self.assertRaises(HTTPException) as e:
+            self.library_manger.install_library(self.library_uuid, BytesIO())
+        self.assertEqual(e.exception.detail, "不正なZIPファイルです。")
+
+        invalid_vvlib_name = "test/invalid.vvlib"
+        self.create_vvlib_without_manifest(invalid_vvlib_name)
+
+        with open(invalid_vvlib_name, "br") as f, self.assertRaises(HTTPException) as e:
+            self.library_manger.install_library(self.library_uuid, f)
+        self.assertEqual(e.exception.detail, "指定された音声ライブラリにvvlib_manifest.jsonが存在しません。")
+
+        self.create_vvlib_without_manifest(invalid_vvlib_name)
+        with ZipFile(invalid_vvlib_name, "a") as zf:
+            zf.writestr(vvlib_manifest_name, "test")
+
+        with open(invalid_vvlib_name, "br") as f, self.assertRaises(HTTPException) as e:
+            self.library_manger.install_library(self.library_uuid, f)
+        self.assertEqual(e.exception.detail, "指定された音声ライブラリのvvlib_manifest.jsonは不正です。")
+
+        invalid_vvlib_manifest = copy.deepcopy(self.vvlib_manifest)
+        invalid_vvlib_manifest["version"] = 10
+        # Duplicate name: 'vvlib_manifest.json'とWarningを吐かれるので、作り直す
+        self.create_vvlib_without_manifest(invalid_vvlib_name)
+        with ZipFile(invalid_vvlib_name, "a") as zf:
+            zf.writestr(vvlib_manifest_name, json.dumps(invalid_vvlib_manifest))
+
+        with open(invalid_vvlib_name, "br") as f, self.assertRaises(HTTPException) as e:
+            self.library_manger.install_library(self.library_uuid, f)
+        self.assertEqual(
+            e.exception.detail, "指定された音声ライブラリのvvlib_manifest.jsonに不正なデータが含まれています。"
+        )
+
+        invalid_vvlib_manifest["version"] = "10"
+        # Duplicate name: 'vvlib_manifest.json'とWarningを吐かれるので、作り直す
+        self.create_vvlib_without_manifest(invalid_vvlib_name)
+        with ZipFile(invalid_vvlib_name, "a") as zf:
+            zf.writestr(vvlib_manifest_name, json.dumps(invalid_vvlib_manifest))
+
+        with open(invalid_vvlib_name, "br") as f, self.assertRaises(HTTPException) as e:
+            self.library_manger.install_library(self.library_uuid, f)
+        self.assertEqual(e.exception.detail, "指定された音声ライブラリのversionが不正です。")
+
+        invalid_vvlib_manifest["version"] = "0.0.1"
```

Comment:
どこらへんで次のテスト内容になってるのかちょっとわかりにくいので、簡単にコメント書いていくのどうでしょう。
面倒であればとりあえず改行を２行ずつにするとかでも。
```suggestion
        # 不正なマニフェストバージョン
        invalid_vvlib_manifest["version"] = "0.0.1"
```

Reply:
確かに分かりづらかったですね...コメントを書いておきました！